### PR TITLE
feat: strict

### PR DESCRIPTION
**Pull request by bodinsamuel**
- Add a lot of missing rules, try to explain why some of them are disabled (to avoid loosing track of the decision)
- Support for jsdoc #121 
- Support for jsx a11y
- Add test for typescript, comments, jsx a11y

### CODE REVIEW

File: packages/eslint-config-algolia/rules/base.js [hunk 1]
```diff
@@ -327,5 +342,43 @@ module.exports = {
...
+      'error',
+      {
+        definedTags: ['required', 'swagger'],
+      },
+    ],
+    'jsdoc/check-types': ['error'],
+    'jsdoc/implements-on-classes': ['error'],
+    'jsdoc/match-description': ['error'],
+    'jsdoc/newline-after-description': ['error'],
+    'jsdoc/no-types': ['error'],
```
Comment (Haroenv) [inline packages/eslint-config-algolia/rules/base.js:371]: should this maybe only be used in typescript?

File: packages/eslint-config-algolia/rules/react.js [hunk 2]
```diff
@@ -10,7 +10,112 @@ module.exports = {
...
+    'react-hooks/rules-of-hooks': ['error'],
+    'react-hooks/exhaustive-deps': ['warn'],
+
+    'react/button-has-type': ['error'],
+    'react/destructuring-assignment': ['off'],
+    'react/jsx-boolean-value': ['error', 'always'],
+    'react/jsx-filename-extension': ['error', { extensions: ['.tsx', '.jsx'] }],
+    'react/jsx-pascal-case': ['error'],
+    'react/jsx-no-target-blank': ['error'],
+    'react/jsx-wrap-multilines': [
```
Comment (Haroenv) [inline packages/eslint-config-algolia/rules/react.js:22]: doesn't prettier already do this?

File: packages/eslint-config-algolia/rules/react.js [hunk 3]
```diff
@@ -10,7 +10,112 @@ module.exports = {
...
+    ],
+    'react/jsx-props-no-spreading': ['off'],
+    'react/jsx-fragments': ['error'],
+    'react/jsx-handler-names': ['error'],
+    'react/no-access-state-in-setstate': ['error'],
+    'react/no-unescaped-entities': ['error'],
+    'react/no-unused-state': ['error'],
+    'react/prop-types': ['off'],
+    'react/sort-comp': [
+      'off',
```
Comment (Haroenv) [inline packages/eslint-config-algolia/rules/react.js:42]: if it's off, why are options given?

File: packages/eslint-config-algolia/rules/react.js [hunk 4]
```diff
@@ -10,7 +10,112 @@ module.exports = {
...
+    'jsx-a11y/media-has-caption': ['error'],
+    'jsx-a11y/mouse-events-have-key-events': ['error'],
+    'jsx-a11y/no-access-key': ['error'],
+    'jsx-a11y/no-autofocus': ['error', { ignoreNonDOM: true }],
+    'jsx-a11y/no-distracting-elements': [
+      'error',
+      {
+        elements: ['marquee', 'blink'],
+      },
+    ],
```
Comment (Haroenv) [inline packages/eslint-config-algolia/rules/react.js:108]: do we really need this rule? blink doesn't even exist anymore

File: packages/eslint-config-algolia/rules/typescript.js [hunk 5]
```diff
@@ -1,11 +1,58 @@
...
+            fixWith: 'Record<string, unknown>',
+          },
+          Function: 'Use a specific function type instead, like `() => void`.',
+        },
+      },
+    ],
+    '@typescript-eslint/consistent-type-imports': [
+      'error',
+      { prefer: 'type-imports' },
+    ],
```
Comment (Haroenv) [inline packages/eslint-config-algolia/rules/typescript.js:55]: I'm not sure if this one is needed, what's the advantage of splitting up types & other imports?

File: packages/eslint-config-algolia/rules/base.js (hunk 1 above)
Comment (bodinsamuel) [reply packages/eslint-config-algolia/rules/base.js:371]: indeed 👌🏻

File: packages/eslint-config-algolia/rules/react.js (hunk 2 above)
Comment (bodinsamuel) [reply packages/eslint-config-algolia/rules/react.js:22]: mmh I'm not sure with this level of details

File: packages/eslint-config-algolia/rules/react.js (hunk 3 above)
Comment (bodinsamuel) [reply packages/eslint-config-algolia/rules/react.js:42]: true

File: packages/eslint-config-algolia/rules/react.js (hunk 4 above)
Comment (bodinsamuel) [reply packages/eslint-config-algolia/rules/react.js:108]: it's the recommended settings

File: packages/eslint-config-algolia/rules/typescript.js (hunk 5 above)
Comment (bodinsamuel) [reply packages/eslint-config-algolia/rules/typescript.js:55]: types import can be safely discarded by compilation without having to compute if they are required at runtime.
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html

Comment (Haroenv) [reply packages/eslint-config-algolia/rules/typescript.js:55]: sounds good!

**Review by Haroenv — APPROVED** [review]
good as a major version (since it changes the rules)

**Comment by bodinsamuel** [comment]
Thanks @Haroenv

**Pull request merged**